toolchain/check/impl_lookup.cpp

```diff
@@ -1088,6 +1210,57 @@ auto EvalLookupSingleImplWitness(Context& context, SemIR::LocId loc_id,
   }
 }
 
+auto EvalLookupSingleImplWitness(Context& context, SemIR::LocId loc_id,
```

Comment:
The refactoring of EvalLookupSingleImplWitness doesn't immediately appear to be related to this PR, can it be done separately? All we need to do here is add a call to `LookupDestroyCustomWitness` in it, and add the result to the cache before returning a MakeFinal.